applications/ShapeOptimizationApplication/python_scripts/algorithm_factory.py

```diff
@@ -11,32 +11,39 @@
 # Making KratosMultiphysics backward compatible with python 2.6 and 2.7
 from __future__ import print_function, absolute_import, division
 
-# Additional imports
-from .algorithm_steepest_descent import AlgorithmSteepestDescent
-from .algorithm_penalized_projection import AlgorithmPenalizedProjection
-from .algorithm_trust_region import AlgorithmTrustRegion
-from .algorithm_bead_optimization import AlgorithmBeadOptimization
 
 # ==============================================================================
 def CreateOptimizationAlgorithm(optimization_settings, analyzer, communicator, model_part_controller):
     algorithm_name = optimization_settings["optimization_algorithm"]["name"].GetString()
 
     if algorithm_name == "steepest_descent":
+        from .algorithm_steepest_descent import AlgorithmSteepestDescent
         return AlgorithmSteepestDescent(optimization_settings,
                                         analyzer,
                                         communicator,
                                         model_part_controller)
+    elif algorithm_name == "gradient_projection":
+        from .algorithm_gradient_projection import AlgorithmGradientProjection
+        if not AlgorithmGradientProjection:
+            raise RuntimeError("'gradient_projection' algorithm is not available. Hint: 'EigenSolversApplication' is required.")
+        return AlgorithmGradientProjection(optimization_settings,
```

Comment:
```suggestion
        from .algorithm_gradient_projection import AlgorithmGradientProjection
        return AlgorithmGradientProjection(optimization_settings,
```

this is not necessary, it won't even reach this point